.i18n/zh_CN.po

```diff
@@ -0,0 +1,5296 @@
+"not the proof. If `h : P = Q` is the proof, then `rw [h]` will work.\n"
+"\n"
+"## Details\n"
+"\n"
+"The `rw` tactic is a way to do \"substituting in\". There\n"
+"are two distinct situations where you can use this tactic.\n"
+"\n"
+"1) Basic usage: if `h : A = B` is an assumption or\n"
+"the proof of a theorem, and if the goal contains one or more `A`s, then `rw [h]`\n"
+"will change them all to `B`'s. The tactic will error\n"
+"if there are no `A`s in the goal.\n"
+"\n"
+"2) Advanced usage: Assumptions coming from theorem proofs\n"
+"often have missing pieces. For example `add_zero`\n"
+"is a proof that `? + 0 = ?` because `add_zero` really is a function,\n"
+"and `?` is the input. In this situation `rw` will look through the goal\n"
+"for any subterm of the form `x + 0`, and the moment it\n"
+"finds one it fixes `?` to be `x` then changes all `x + 0`s to `x`s.\n"
+"\n"
+"Exercise: think about why `rw [add_zero]` changes the term\n"
+"`(0 + 0) + (x + 0) + (0 + 0) + (x + 0)` to\n"
+"`0 + (x + 0) + 0 + (x + 0)`\n"
+"\n"
+"If you can't remember the name of the proof of an equality, look it up in\n"
+"the list of lemmas on the right.\n"
+"\n"
+"## Targetted usage\n"
+"\n"
+"If your goal is `b + c + a = b + (a + c)` and you want to rewrite `a + c`\n"
+"to `c + a`, then `rw [add_comm]` will not work because Lean finds another\n"
+"addition first and swaps those inputs instead. Use `rw [add_comm a c]` to\n"
+"guarantee that Lean rewrites `a + c` to `c + a`. This works because\n"
+"`add_comm` is a proof that `?1 + ?2 = ?2 + ?1`, `add_comm a` is a proof\n"
+"that `a + ? = ? + a`, and `add_comm a c` is a proof that `a + c = c + a`.\n"
+"\n"
+"If `h : X = Y` then `rw [h]` will turn all `X`s into `Y`s.\n"
+"If you only want to change the 37th occurrence of `X`\n"
+"to `Y` then do `nth_rewrite 37 [h]`."
+msgstr ""
+"## 摘要\n"
+"\n"
+"如果 `h` 是等式 `X = Y` 的证明，那么 `rw [h]` 将改变\n"
+"目标中的所有 `X`s 变为 `Y`s。这是 \"代入 \"的方法。\n"
+"\n"
+"## Variants\n"
+"\n"
+"* `rw [← h]` （将 `Y`s 更改为 `X`s；通过输入 `\\left ` 或 `\\l` 获取后箭头）。\n"
+"\n"
+"* `rw [h1, h2]`（重写序列）\n"
+"\n"
+"* `rw [h] at h2`（将假设 `h2` 中的 `X`s 改为 `Y`s）\n"
+"\n"
+"* `rw [h] at h1 h2 ⊢` （将两个假设和目标中的 `X`s 改为 `Y`s；\n"
+"用 `\\|-` 获取 `⊢` 符号）。\n"
+"\n"
+"* `repeat rw [add_zero]` 将继续将 `? + 0` 更改为 `?`。\n"
+"直到没有更多匹配的 `? + 0`。\n"
+"\n"
+"* `nth_rewrite 2 [h]` 只会将目标中的第二个 `X` 改为 `Y`。\n"
+"\n"
+"#### 示例：\n"
+"\n"
+"如果假设为 `h : x = y + y`，目标为\n"
+"```.\n"
+"succ (x + 0) = succ (y + y)\n"
+"```.\n"
+"\n"
+"则\n"
+"\n"
+"`rw [add_zero]`\n"
+"\n"
+"会将目标改为 `succ x = succ (y + y)`，然后\n"
+"\n"
+"`rw [h]`\n"
+"\n"
+"会将目标变为 `succ (y + y) = succ (y + y)`，这\n"
+"可以用 `rfl` 解决。\n"
+"\n"
+"#### 示例：\n"
+"\n"
+"你也可以用 `rw` 来改变一个假设。\n"
+"例如，如果您有两个假设\n"
+"```\n"
+"h1 : x = y + 3\n"
+"h2 : 2 * y = x\n"
+"```\n"
+"则 `rw [h1] at h2` 将把 `h2` 变为 `h2 : 2 * y = y + 3`。\n"
+"-/\n"
+"\n"
+"## 常见错误\n"
+"\n"
+"* 需要方括号。`rw h` 永远不会正确。\n"
+"\n"
+"* 如果 `h` 不是一个 *equality* 的 *proof* （形式为 `A = B` 的语句）、\n"
```

Comment:
*equality* and *proof* are not translated. Should be `*等式* 的 *证明*`.

Reply:
Thanks for reviewing!

@kbuzzard is it generally okay for you to (more or less blindly) trust translations (and add them) if we don't speak the target language? (Currently the translation is not available in-game, but once the front-end has been implemented, you should be able to select your language.)

Reply:
> _equality_ and _proof_ are not translated. Should be `*等式* 的 *证明*`.

fixed

Reply:
My two cents:

1. The current translation seems to be mostly machine translated, has strong "translation tone"(翻译腔), and shows little sign of manual corrections (Disclaimer: I'm a native Chinese speaker);
2. It could be considered an initial translation for others to work on and improve, so accepting the PR could be considered, or it could be more polished in this PR before merging;
3. In general, contributors of translations could be asked to submit the following information alongside the translation:

> Translation method: Completely machine, machine assisted, manual?
> Translation review process: has the translation been reviewed by a small group of native speakers, and what are the feedbacks?
> Initial target audience: is this for a course targeting a certain number of students, or just sharing between some friends etc. but maybe thery are research mathematians?
> Translation maintainance: is the contributor willing to continously gather feedbacks and fix the translations, at least for a few months?

Then a reviewer can be more certain about whether a specific PR should be merged.

Reply:
Thanks @utensil that's very helpful! I updated the README with some points you suggested. I'm quite fine with the quality of translations in general, but having some external review like you just did should be preferrable:

> We happily accept translations of the game into different lanugages! You can use `.i18n/Game.pot` and translate it into `.i18n/Game-{lang}.po` where `{lang}` is the ISO language code like `fr` or `en_UK`, using for example POEdit.
> 
> We would like the following requirements for a translation PR:
> 
> - One independent person from the community, who understands the language, gives a review on the PR. You could for example look at the [Lean Community Map](https://leanprover-community.github.io/meet.html) or ask on Zulip. Such a review an be quite generic and does not have to be super detailed.
> - In the credits (i.e. in the string translating them), ideally you should add yourself as a translator for this language.

Reply:
> My two cents:
> 
> 1. The current translation seems to be mostly machine translated, has strong "translation tone"(翻译腔), and shows little sign of manual corrections (Disclaimer: I'm a native Chinese speaker);
> 2. It could be considered an initial translation for others to work on and improve, so accepting the PR could be considered, or it could be more polished in this PR before merging;
> 3. In general, contributors of translations could be asked to submit the following information alongside the translation:
> 
> * Translation method: Completely machine, machine assisted, manual? Please elaborate a little on your manual corrections.
> * Translation review process: Has the translation been reviewed by a small group of native speakers, and what are the feedbacks?
> * Initial target audience: Is this for a course targeting a certain number of students, or just sharing between some friends etc. but maybe they are research mathematicians?
> * Translation maintenance: Is the contributor willing to continuously gather feedbacks and fix the translations, at least for a few months?
> 
> Then a reviewer can be more certain about whether a specific PR should be merged.

Translation method: machine assisted
Translation review process:  @utensil  Do you want to review?
Initial target audience: lower-grade undergraduates or high school students. Share in teachers first.
Translation maintenance: Yes, can help for six months.

@utensil  Hi, although the translation was initiated by the machine, I spent more than 30 hours checking each script, fixing them to make them readable. The machine completely lacks awareness that one is both playing a game and learning math simultaneously, so it has no understanding of the context, and its hard to find other similar words in the world. As a result, the initial translation is awful. Additionally, I personally find the translation challenging. Fixing them takes effect. I do not aim to prepare an elegant translation. What I considered is whether a lower-grade undergraduate student or a high school student can understand the script. Therefore, if you notice any parts of the script that are unclear or might be misunderstood by students, please provide direct feedback. 

Additionally, since the scripts are highly fragmented, we should review the final result in the game. This way, we can see how they come together and identify any missing part that may make the game difficult to play

Reply:
Thanks for the explanation, apologies for underestimating your efforts put into the manual fixes, also thanks for your work on i18next.

The proposal looks good to me, previously I only skimmed, I'll try to go over it again later today.

I definitely agree that improvements on the translation tone belongs to separate PRs as it's really non-trivial.

Reply:
> Thanks for the explanation, apologies for underestimating your efforts put into the manual fixes, also thanks for your work on i18next.
> 
> The proposal looks good to me, previously I only skimmed, I'll try to go over it again later today.
> 
> I definitely agree that improvements on the translation tone belongs to separate PRs as it's really non-trivial and quite opinionated.

Thank you. Don't hesitate to give advice, even if it is about tone. I'm pretty sure I've left many inconsistency and maybe significant errors in the script, though I failed to find them when I, myself, review the script.

Reply:
Also note, anybody (also students playing) can later still submit corrections with very little effort, so you really don't need to spent more time on reviewing than you feel is necessary.

Reply:
@JiechengZhao Sorry for the delay, I have finished a more care pass, and submitted the comments, most of them are just suggestions, only few are mistakes.